File: pagamento-oneclick/su-sito-merchant/pagamenti-successivi/tramite-redirezione/dotnet/notifica.cs
                                            
// Pagamento OneClik - Pagamenti successivi - Tramite redirezione - Notifica

using System;
using System.Collections.Generic;
using System.Text;
using System.Security.Cryptography;

namespace XPay
{
    class SampleXPayResponse
    {
        static void Main(string[] args)
        {

            // Chiave segreta
            string CHIAVESEGRETA = "<CHIAVE SEGRETA PER CALCOLO MAC>"; // Sostituire con il valore fornito da Nexi

            // Calcolo MAC
            SampleXPayRequest sampleXPay = new SampleXPayRequest();
            string mac = sampleXPay.HashMac(  "codTrans=" + args.codTrans +
                                              "esito=" + args.esito +
                                              "importo=" + args.importo +
                                              "divisa=" + args.divisa +
                                              "data=" + args.data +
                                              "orario=" + args.orario +
                                              "codAut=" + args.codAut +
                                              CHIAVESEGRETA);

            // Verifico corrispondenza tra MAC calcolato e MAC di ritorno
            if(mac != args.mac)
            {
                Console.WriteLine("Errore MAC: " + mac + " non corrisponde a " + args.mac);
            } else
            {
                // Nel caso in cui non ci siano errori gestisco il parametro esito
                if (args.esito == "OK")
                {
                    Console.WriteLine("OK, pagamento avvenuto, preso riscontro");
                } else
                {
                    Console.WriteLine("KO, pagamento non avvenuto, preso riscontro");
                }
            }


        }
    }
}
                    
                

File: pagamento-oneclick/su-sito-merchant/primo-pagamento/php/notifica.php
<?php

// Chiave segreta
$chiaveSegreta = "<CHIAVE SEGRETA PER CALCOLO MAC>"; // Sostituire con il valore fornito da Nexi

// Controllo che si siano tutti i parametri di ritorno obbligatori per il calcolo del MAC
$requiredParams = array('codTrans', 'esito', 'importo', 'divisa', 'data', 'orario', 'codAut', 'mac');
foreach ($requiredParams as $param) {
    if (!isset($_REQUEST[$param])) {
        echo 'Paramentro mancante ' . $field;
        header('500 Internal Server Error', true, 500);
        exit;
    }
}

// Calcolo MAC con parametri di ritorno
$macCalculated = sha1('codTrans=' . $_REQUEST['codTrans'] .
        'esito=' . $_REQUEST['esito'] .
        'importo=' . $_REQUEST['importo'] .
        'divisa=' . $_REQUEST['divisa'] .
        'data=' . $_REQUEST['data'] .
        'orario=' . $_REQUEST['orario'] .
        'codAut=' . $_REQUEST['codAut'] .
        $chiaveSegreta
);


// Verifico corrispondeza tra MAC calcolato e parametro mac di ritorno
if ($macCalculated != $_REQUEST['mac']) {
    echo 'Errore MAC: ' . $macCalculated . ' non corrisponde a ' . $_REQUEST['mac'];
    header('500 Internal Server Error', true, 500);
    exit;
}

// nel caso in cui non ci siano errori gestisco il parametro esito
if($_REQUEST['esito'] == 'OK'){
    header('OK, pagamento avvenuto, preso riscontro', true, 200);
} else {
    header('KO, pagamento non avvenuto, preso riscontro', true, 200);
}
